orttraining/orttraining/core/graph/adasum_optimizer_graph_builder.cc

```diff
@@ -130,14 +130,24 @@ Status AdasumOptimizerGraphBuilder::BuildInternal(
   // check if all gradients are finite
   ArgDef global_grad_norm_argdef;
   ArgDef global_grad_norm_finite_argdef;
-  ORT_RETURN_IF_ERROR(AddGradientNorm(
-      nodearg_name_generator, gradient_argdefs, graph_defs, global_grad_norm_argdef));
-  optimizer_graph_outputs[OptimizerOutputKey::GlobalGradientNorm] = global_grad_norm_argdef.name;
+
+  if (opt_graph_config_.use_mixed_precision &&
+      opt_graph_config_.mixed_precision_type == MixedPrecisionDataType::BP16) {
+    //gradient norm for bfloat16 is not ready yet. skip it to unblock the testing
+    //will add it back when it is ready
+    ;
+  } else {
+    ORT_RETURN_IF_ERROR(AddGradientNorm(
+        nodearg_name_generator, gradient_argdefs, graph_defs, global_grad_norm_argdef));
+    optimizer_graph_outputs[OptimizerOutputKey::GlobalGradientNorm] = global_grad_norm_argdef.name;
+  }
 
   if (opt_graph_config_.use_mixed_precision) {
-    ORT_RETURN_IF_ERROR(AddFiniteGradientCheck(
-        nodearg_name_generator, {global_grad_norm_argdef}, graph_defs, global_grad_norm_finite_argdef));
-    optimizer_graph_outputs[OptimizerOutputKey::GradientAllIsFinite] = global_grad_norm_finite_argdef.name;
+    if (opt_graph_config_.mixed_precision_type == MixedPrecisionDataType::FP16) {
```

Comment:
[nit] It seems we have too many if-else blocks, but I don't have a quick solution for this PR.